persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/MolecularProfileCaseIdentifierUtil.java

```diff
@@ -0,0 +1,33 @@
+package org.cbioportal.persistence.mybatis.util;
+
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class MolecularProfileCaseIdentifierUtil {
+
+    public Map<String, Set<String>> getGroupedCasesByMolecularProfileId(List<String> molecularProfileIds, List<String> caseIds) {
+        Map<String, Set<String>> groupMolecularProfileSamples = new HashMap<>();
+
+        if (CollectionUtils.isEmpty(caseIds)) {
+            return molecularProfileIds.stream().collect(Collectors.toMap(Function.identity(), e -> new HashSet<>()));
+        }
+
+        for (int i = 0; i < molecularProfileIds.size(); i++) {
+            String molecularProfileId = molecularProfileIds.get(i);
+            String caseId = caseIds.get(i);
```

Comment:
seems slightly odd because it requires caseIds to be of same dimension as molecularProfileIds.  is that gauranteed?

Reply:
True, but this is how it is passed from controller layer. Similar style is all over the place(in other api's). Lets handle it in separately. This logic does work